### PR TITLE
test(sbb-overlay): add visual tests

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**

### CODE REVIEW

File: src/elements/overlay/overlay.visual.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+describe(`sbb-overlay`, () => {
+  const defaultArgs = {
+    expanded: false,
+    backButton: false,
+    negative: false,
+  };
+
+  const template = ({ expanded, backButton, negative }: typeof defaultArgs): TemplateResult => html`
+    <sbb-overlay ?negative=${negative} ?expanded=${expanded} ?back-button=${backButton}>
+      <div class="overlay-content">
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.visual.spec.ts:17]: is the div and the class needed?

File: src/elements/overlay/overlay.visual.spec.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
+        labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
+        laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
+        voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
+        cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+      </div>
+    </sbb-overlay>
+  `;
+
+  const openOverlay = (setup: any): void => {
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.visual.spec.ts:27]: Could it be typed instead of any? :/

File: src/elements/overlay/overlay.visual.spec.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    const overlay: SbbOverlayElement = setup.snapshotElement.querySelector('sbb-overlay')!;
+    setup.withSnapshotElement(overlay);
+    setup.withPostSetupAction(() => overlay.open());
+  };
+
+  describeViewports({ viewportHeight: 600 }, () => {
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.visual.spec.ts:33]: does it make sense to test every viewport, or would a subset be fine?

File: src/elements/overlay/overlay.visual.spec.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    for (const negative of [false, true]) {
+      it(
+        `negative=${negative} ${visualDiffDefault.name}`,
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.visual.spec.ts:36]:
```suggestion
        `negative=${negative}`,
```

File: src/elements/overlay/overlay.visual.spec.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        visualDiffDefault.with(async (setup) => {
+          await setup.withFixture(template({ ...defaultArgs, negative }));
+          openOverlay(setup);
+        }),
+      );
+
+      it(
+        `negative=${negative} backButton ${visualDiffDefault.name}`,
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.visual.spec.ts:44]:
```suggestion
        `negative=${negative} backButton`,
```

File: src/elements/overlay/overlay.visual.spec.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        visualDiffDefault.with(async (setup) => {
+          await setup.withFixture(template({ ...defaultArgs, negative, backButton: true }));
+          openOverlay(setup);
+        }),
+      );
+
+      it(
+        `negative=${negative} expanded ${visualDiffDefault.name}`,
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.visual.spec.ts:52]:
```suggestion
        `negative=${negative} expanded`,
```

File: src/elements/overlay/overlay.visual.spec.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import { html, type TemplateResult } from 'lit';
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.visual.spec.ts:1]: Would it be possible to have a scenario where the scrollbars are visible?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

File: src/elements/overlay/overlay.visual.spec.ts (hunk 2 above)
Comment (DavideMininni-Fincons) [reply src/elements/overlay/overlay.visual.spec.ts:27]: No, but we can consider to export the VisualDiffSetupBuilder class

File: src/elements/overlay/overlay.visual.spec.ts (hunk 3 above)
Comment (DavideMininni-Fincons) [reply src/elements/overlay/overlay.visual.spec.ts:33]: On Figma i see every viewport, but if required we can reduce it (eg. zero/small/large/ultra ?)

File: src/elements/overlay/overlay.visual.spec.ts (hunk 2 above)
Comment (jeripeierSBB) [reply src/elements/overlay/overlay.visual.spec.ts:27]: @kyubisation What do you think? I remember you preferred to have it not exported.

File: src/elements/overlay/overlay.visual.spec.ts (hunk 3 above)
Comment (jeripeierSBB) [reply src/elements/overlay/overlay.visual.spec.ts:33]: I found styles for zero, small, medium, large and ultra. so maybe keeping all is ok ;-)

File: src/elements/overlay/overlay.visual.spec.ts (hunk 2 above)
Comment (kyubisation) [reply src/elements/overlay/overlay.visual.spec.ts:27]: Yes, but purely due to not having a necessity. If it helps in this case, then we can export it.

Comment (jeripeierSBB) [reply src/elements/overlay/overlay.visual.spec.ts:27]: ok, thanks! So I'd prefer exporting it over having `any` type

File: src/elements/overlay/overlay.visual.spec.ts (hunk 7 above)
Comment (DavideMininni-Fincons) [reply src/elements/overlay/overlay.visual.spec.ts:1]: Done

File: src/elements/overlay/overlay.visual.spec.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+          sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id
+          est laborum. <br />
+        `,
+      )}
+    </sbb-overlay>
+  `;
+
+  const openOverlay = (setup: VisualDiffSetupBuilder): void => {
+    const overlay: SbbOverlayElement = setup.snapshotElement.querySelector('sbb-overlay')!;
+    setup.withSnapshotElement(overlay);
```
Comment (jeripeierSBB) [inline src/elements/overlay/overlay.visual.spec.ts:41]: Should not be needed
```suggestion
```

**Review by jeripeierSBB — APPROVED** [review]
LGTM, probably one redundant line

**Pull request merged**